pkg/helpers/newapp/app/imageref.go

```diff
@@ -230,16 +230,27 @@ func (r *ImageRef) SuggestName() (string, bool) {
 	if r == nil {
 		return "", false
 	}
+	var name string
 	// if ImageStream already exists, its name should
 	// have higher preference than object name
-	if r.Stream != nil {
-		return r.Stream.Name, true
-	}
-	if len(r.ObjectName) > 0 {
-		return r.ObjectName, true
+	switch {
+	case r.Stream != nil:
+		name = r.Stream.Name
+	case len(r.ObjectName) > 0:
+		name = r.ObjectName
+	case len(r.Reference.Name) > 0:
+		name = r.Reference.Name
 	}
-	if len(r.Reference.Name) > 0 {
-		return r.Reference.Name, true
+	if len(name) > 0 {
+		if r.AsImageStream {
+			// in some certain situations, suggested ImageStream name might contain slashes
+			// in this case we want to replace it with dashes, referenced object might
+			// contain said dashes, but imagestream must cannot
+			// example: https://bugzilla.redhat.com/show_bug.cgi?id=1970805
+			name = strings.ReplaceAll(name, "/", "-")
+		}
+
+		return name, true
 	}
 	return "", false
```

Comment:
I think that we could eliminate the `if len(name)) > 0` above and just return `name, len(name) > 0` here?
```suggestion
	return name, len(name) > 0
```